fix(network): string type in node/linkComponent

jraymakers opened the pull request:
Fixes #939 

#### :boom: Breaking Changes

- This changes the type signature of a public-facing prop, so it might be considered a breaking change. However, the type option that's being removed was broken anyway, so this doesn't break any functionality.

#### :memo: Documentation

- Extended `network` example to show use of `nodeComponent` and `linkComponent`, as well as `top` and `left` props. Adapted from demo in comment of https://github.com/airbnb/visx/issues/939

#### :bug: Bug Fix

- Remove string type from `nodeComponent` and `linkComponent` props in `network` package. This type was invalid in spite of being in the type signature. See https://github.com/airbnb/visx/issues/939

jraymakers commented:
I can't tell why Happo failed, because I can't follow the Detail links. But, it seems likely it's detecting the changes I made (intentionally) to the `network` gallery page.

williaster commented:
Hey @jraymakers sorry for the delay, thanks for the fix! I agree this shouldn't be considered breaking since it was broken :)

We recently added happo and there are a couple of demos that are flaky, apologies for making you think you did something wrong and that you can't access the report. This is what it's reporting for your change: 

![image](https://user-images.githubusercontent.com/4496521/108926884-1e8d9f80-75f4-11eb-80e7-709b0b375e69.png)

It's a pretty big change aesthetically, I'm wondering if we could update the logic such that a single link is changed and a single node is changed. This would demonstrate the customizability and keep the aesthetics similar to before.

jraymakers commented:
Sure, that makes sense. I'll make the customizations more subtle.

jraymakers commented:
@williaster I changed the example to just have one custom node and one custom link, and used styling that should fit in better with the defaults and the other gallery tiles. I also fixed the clipping issue.

As an added bonus, the example now shows how to use custom node and link types with this generic component.

williaster reviewed (approved):
@jraymakers amazing, it looks great! thanks for being open to iterating on styles 😄 

![image](https://user-images.githubusercontent.com/4496521/109053606-be4d3b00-7691-11eb-9a93-7047630cc778.png)

Heads up I'm working on automating releases, I think this could be a good candidate for testing but it may take me a couple days to get it right.
